Error: Building package libepoxy:x86-windows failed with: BUILD_FAILED

I performed ".\vcpkg update" and then ".\vcpkg install gtk". The output was:

>The following packages will be built and installed:
>    gtk:x86-windows
>  \* libepoxy:x86-windows
>Additional packages (*) will be installed to complete this operation.
>Building package libepoxy:x86-windows...
>-- CURRENT_INSTALLED_DIR=C:/Program Files (x86)/vcpkg/vcpkg/installed/x86-windows
>-- DOWNLOADS=C:/Program Files (x86)/vcpkg/vcpkg/downloads
>-- CURRENT_PACKAGES_DIR=C:/Program Files (x86)/vcpkg/vcpkg/packages/libepoxy_x86-windows
>-- CURRENT_BUILDTREES_DIR=C:/Program Files (x86)/vcpkg/vcpkg/buildtrees/libepoxy
>-- CURRENT_PORT_DIR=C:/Program Files (x86)/vcpkg/vcpkg/ports/libepoxy/.
>-- Using cached C:/Program Files (x86)/vcpkg/vcpkg/downloads/anholt-libepoxy-1.4.3.tar.gz
>-- Testing integrity of cached file...
>-- Testing integrity of cached file... OK
>-- Extracting done
>-- Configuring x86-windows-rel
>CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:43 (message):
>    Command failed: C:/Program Files (x86)/vcpkg/vcpkg/downloads/tools/python/python.exe;C:/Program Files (x86)/vcpkg/vcpkg/downloads/tools/meson/meson-0.40.1/meson.py;-Denable-glx=no;-Denable-egl=no;--buildtype;plain;--backend;ninja;--default-library;shared;--prefix;C:/Program Files (x86)/vcpkg/vcpkg/packages/libepoxy_x86-windows;C:/Program Files (x86)/vcpkg/vcpkg/buildtrees/libepoxy/src/libepoxy-1.4.3
>    Working Directory: C:/Program Files (x86)/vcpkg/vcpkg/buildtrees/libepoxy/x86-windows-rel
>    See logs for more information:
>      C:\Program Files (x86)\vcpkg\vcpkg\buildtrees\libepoxy\config-x86-windows-rel-out.log
>      C:\Program Files (x86)\vcpkg\vcpkg\buildtrees\libepoxy\config-x86-windows-rel-err.log
>
>Call Stack (most recent call first):
>  scripts/cmake/vcpkg_configure_meson.cmake:51 (vcpkg_execute_required_process)
>  ports/libepoxy/portfile.cmake:14 (vcpkg_configure_meson)
>  scripts/ports.cmake:72 (include)
>
>
>Error: Building package libepoxy:x86-windows failed with: BUILD_FAILED
>Please ensure you're using the latest portfiles with `.\vcpkg update`, then
>submit an issue at https://github.com/Microsoft/vcpkg/issues including:
>  Package: libepoxy:x86-windows
>  Vcpkg version: 0.0.82-f2c655c7dfa02fe0fcca4a6207f3f1cc55da540c
>
>Additionally, attach any relevant sections from the log files above.

Please see the attached logs. Let me know if there's any more information I can provide.
[config-x86-windows-rel-err.log.txt](https://github.com/Microsoft/vcpkg/files/1217165/config-x86-windows-rel-err.log.txt)
[config-x86-windows-rel-out.log.txt](https://github.com/Microsoft/vcpkg/files/1217166/config-x86-windows-rel-out.log.txt)

--- a/toolsrc/include/vcpkg_GlobalState.h
+++ b/toolsrc/include/vcpkg_GlobalState.h
@@ -12,5 +12,8 @@ namespace vcpkg
         static Util::LockGuarded<ElapsedTime> timer;
         static std::atomic<bool> debugging;
         static std::atomic<bool> feature_packages;
+
+        static std::atomic<int> g_init_console_cp;
+        static std::atomic<int> g_init_console_output_cp;
     };
 }

--- a/toolsrc/src/vcpkg.cpp
+++ b/toolsrc/src/vcpkg.cpp
@@ -18,30 +18,6 @@
 
 using namespace vcpkg;
 
-UINT console_cp_input;
-UINT console_cp_output;
-
-void console_cp_set()
-{
-    console_cp_input = GetConsoleCP();
-    console_cp_output = GetConsoleOutputCP();
-
-    SetConsoleCP(CP_UTF8);
-    SetConsoleOutputCP(CP_UTF8);
-}
-
-void console_cp_reset()
-{
-    SetConsoleCP(console_cp_input);
-    SetConsoleOutputCP(console_cp_output);
-}
-
-BOOL console_ctrl_handler(DWORD fdwCtrlType)
-{
-    console_cp_reset();
-    return TRUE;
-}
-
 void invalid_command(const std::string& cmd)
 {
     System::println(System::Color::error, "invalid command: %s", cmd);
@@ -216,13 +192,13 @@ int wmain(const int argc, const wchar_t* const* const argv)
 {
     if (argc == 0) std::abort();
 
-    console_cp_set();
-    SetConsoleCtrlHandler(PHANDLER_ROUTINE(console_ctrl_handler), TRUE);
-    atexit(console_cp_reset);
+    GlobalState::g_init_console_cp = GetConsoleCP();
+    GlobalState::g_init_console_output_cp = GetConsoleOutputCP();
+
+    SetConsoleCP(65001);
+    SetConsoleOutputCP(65001);
 
     *GlobalState::timer.lock() = ElapsedTime::create_started();
-
-    // Checks::register_console_ctrl_handler();
 
     const std::string trimmed_command_line = trim_path_from_command_line(Strings::to_utf8(GetCommandLineW()));
 
@@ -240,7 +216,7 @@ int wmain(const int argc, const wchar_t* const* const argv)
     if (auto p = args.sendmetrics.get()) Metrics::g_metrics.lock()->set_send_metrics(*p);
     if (auto p = args.debug.get()) GlobalState::debugging = *p;
 
-    vcpkg::Checks::register_console_ctrl_handler();
+    Checks::register_console_ctrl_handler();
 
     if (GlobalState::debugging)
     {

--- a/toolsrc/src/vcpkg.cpp
+++ b/toolsrc/src/vcpkg.cpp
@@ -18,6 +18,30 @@
 
 using namespace vcpkg;
 
+UINT console_cp_input;
+UINT console_cp_output;
+
+void console_cp_set()
+{
+    console_cp_input = GetConsoleCP();
+    console_cp_output = GetConsoleOutputCP();
+
+    SetConsoleCP(CP_UTF8);
+    SetConsoleOutputCP(CP_UTF8);
+}
+
+void console_cp_reset()
+{
+    SetConsoleCP(console_cp_input);
+    SetConsoleOutputCP(console_cp_output);
+}
+
+BOOL console_ctrl_handler(DWORD fdwCtrlType)
+{
+    console_cp_reset();
+    return TRUE;
+}
+
 void invalid_command(const std::string& cmd)
 {
     System::println(System::Color::error, "invalid command: %s", cmd);
@@ -191,6 +215,10 @@ static std::string trim_path_from_command_line(const std::string& full_command_l
 int wmain(const int argc, const wchar_t* const* const argv)
 {
     if (argc == 0) std::abort();
+
+    console_cp_set();
+    SetConsoleCtrlHandler(PHANDLER_ROUTINE(console_ctrl_handler), TRUE);
+    atexit(console_cp_reset);
 
     *GlobalState::timer.lock() = ElapsedTime::create_started();
 

--- a/toolsrc/src/vcpkg_Checks.cpp
+++ b/toolsrc/src/vcpkg_Checks.cpp
@@ -17,6 +17,9 @@ namespace vcpkg::Checks
         GlobalState::debugging = false;
         metrics->flush();
 
+        SetConsoleCP(GlobalState::g_init_console_cp);
+        SetConsoleOutputCP(GlobalState::g_init_console_output_cp);
+
         ::exit(exit_code);
     }
 

--- a/toolsrc/src/vcpkg_GlobalState.cpp
+++ b/toolsrc/src/vcpkg_GlobalState.cpp
@@ -7,4 +7,7 @@ namespace vcpkg
     Util::LockGuarded<ElapsedTime> GlobalState::timer;
     std::atomic<bool> GlobalState::debugging = false;
     std::atomic<bool> GlobalState::feature_packages = false;
+
+    std::atomic<int> GlobalState::g_init_console_cp = 0;
+    std::atomic<int> GlobalState::g_init_console_output_cp = 0;
 }

